torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -770,6 +773,460 @@ def propagate_meta_info(new_hop_gm, new_call_function_node, old_call_function_no
     return joint_gm
 
 
+def maybe_log_graph(
+    gm,
+    graph_name,
+    aot_config,
+    structured_log_prefix_fn,
+    out_structured_logs: Optional[list[str]] = None,
+):
+    if not aot_config.enable_log:
+        return
+    aot_graphs_log.debug(
+        "%s",
+        lazy_format_graph_code(
+            f"{graph_name}",
+            gm,
+            aot_config.aot_id,
+            include_stride=True,
+            include_device=True,
+            colored=True,
+        ),
+    )
+    gm_str = gm.print_readable(
+        print_output=False, include_stride=True, include_device=True
+    )
```

Comment:
oh - technically, I think the reason the `payload_fn` of `trace_structured` accepts a lambda is so that if you are running compile without tlparse turned on, you don't need to pay the cost of generating these strings. But this line here kind of defeats the purpose of that, since you are ucnonditionally generating the string here.

I would probably move the `print_readable()` call here directly inside of the lambda.